src/communication/Publisher.cpp

```diff
@@ -10,3 +10,32 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/communication/Publisher.h"
+#include <up-cpp/datamodel/builder/UMessage.h>
+
+namespace uprotocol::communication {
+using namespace uprotocol::datamodel::builder;
+
+Publisher::Publisher(std::shared_ptr<transport::UTransport> transport,
+          v1::UUri& topic, v1::UPayloadFormat format,
+          std::optional<v1::UPriority> priority,
+          std::optional<std::chrono::milliseconds> ttl) :
+    transport_(std::move(transport)) {
+
+    auto messageBuilder = UMessageBuilder::publish(std::move(topic))
+        .withPayloadFormat(format)
+        .withPriority(priority.value_or(v1::UPriority::UPRIORITY_CS1))
+        .withTtl(ttl.value());
```

Comment:
This will throw an exception if ttl is not set. This line would need to be wrapped in an if:
```suggestion
        .withPriority(priority.value_or(v1::UPriority::UPRIORITY_CS1));

    if (ttl) {
        publish_builder_.withTtl(ttl.value());
    }
```

Reply:
done